README.md

```diff
@@ -58,9 +58,49 @@ Do either of the following:
 -   Move them to `/usr/local/lib` or `/usr/lib`.
 -   Set the `LIBRARY_PATH` environment variable to the directory containing the libraries.
 
-### Execution
+### Compilation (static libraries)
+
+For iOS development you have to use static libraries. Get the static libraries from the [vosk-api][vosk-api-ios] team.
+
+#### macOS-only
+The Rust linker is not able to extract the correct architecture from a static fat file itself. Therefore, for each architecture we need a non-fat file to link with our code.
+
+Example for aarch64:
+```shell
+lipo -info libvosk.a
+# Architectures in the fat file: libvosk.a are: armv7 armv7s arm64 
+lipo libvosk.a -thin arm64 -output arm64/libvosk.a
+lipo -info arm64/libvosk.a
+# Non-fat file: arm64/libvosk.a is architecture: arm64
+```
+A typical use case is to create a lib using Rust to integrate with an iOS App. So the next step is to add `staticlib` as crate-type in the `Cargo.toml` file.
+The static Vosk-API libraries must be discoverable and linked by the Rust linker. One very feasible approach is to create a [build script][build-script-explanation].
+
+Example build script (build.rs file) for cross compiling aarch64 Android and iOS:
+```rust
+fn add_lib(location: impl AsRef<str>, name: impl AsRef<str>, static_link: bool) {
+    println!("cargo:rustc-link-search={}", location.as_ref());
+    println!(
+        "cargo:rustc-link-lib={}{}",
+        if static_link { "static=" } else { "" },
+        name.as_ref()
+    )
+}
+
+fn main() {
+    let target =  env::var("TARGET").unwrap();
+    match target.as_str() {
+        "aarch64-apple-ios" => add_lib("./ios-libs/ios-device/arm64", "vosk", true),
+        "aarch64-linux-android" => add_lib("./android-libs/arm64-v8a", "vosk", false),
+        _ => {}
+    }
+}
+```
+In real-world scenarios, one will use Rust to cross compile a library (e.g. Android and iOS). Therefore, we need both `cdylib` as well as the `staticlib` as crate-type. If you compile as usual with cargo build (e.g.: `cargo build --target aarch64-apple-ios --release`) it will not work, because cargo tries to build the dylib as well. Fortunately, since rust 1.64. there is a new option for [rustc](https://github.com/rust-lang/cargo/issues/10083) in the stable channel. Because of this, the following will work: `cargo rustc --crate-type staticlib --lib --target aarch64-apple-ios --release` 
```

Comment:
It's better to keep any android-related explanations separate from the iOS ones. Maybe make another section for it if you're also familiar with android development?

Reply:
I understand what you mean. Is it an option to move the info with crate-type to a troubleshooting subsection? it will be a common stumbling block. But I am also unsure. It doesn't really fit in here, I agree.

Reply:
Yeah, I think this would fit nicely in a troubleshooting section, good idea